viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -56,32 +56,26 @@ std::atomic<bool> b_continue_session{true};
 ::grpc::Status SLAMServiceImpl::GetPosition(ServerContext *context,
                                             const GetPositionRequest *request,
                                             GetPositionResponse *response) {
-    cartographer::transform::Rigid3d global_pose;
-    {
-        std::lock_guard<std::mutex> lk(viam_response_mutex);
-        global_pose = latest_global_pose;
-    }
-
-    auto pos_vector = global_pose.translation();
-    auto pos_quat = global_pose.rotation();
+    viam_carto_get_position_response* vcgpr;
```

Comment:
I think you want to actually instantiate the struct & then pass a reference to it to the function, otherwise I think the pointer could point to uninitialized memory:
```
viam_carto_get_position_response vcgpr;
GetPosition(&vcgpr);
```